internal/provider/resource_gitlab_runner_test.go

```diff
@@ -0,0 +1,121 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabRunner_basic(t *testing.T) {
+	testAccCheck(t)
+
+	group := testAccCreateGroups(t, 1)[0]
+	//The runner token is not populated on the return from the group create, so re-retrieve it to get the token.
+	group, _, err := testGitlabClient.Groups.GetGroup(group.ID, &gitlab.GetGroupOptions{})
+	if err != nil {
+		t.Fail()
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() {},
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckManagedLicenseDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+				resource "gitlab_runner" "this" {
+					token = "%s"
+					description = "Lorem Ipsum"
+				}
+				`, group.RunnersToken),
```

Comment:
Let's add a check if the `authentication_token` has been set using `resource.TestCheckResourceAttrSet()`. At the moment it's not verified, because it's not part of the import ...